test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,250 @@
+# All rights reserved.
+
+require 'minitest_helper'
+require './lib/solarwinds_apm/config'
+require './lib/solarwinds_apm/support/otlp_endpoint'
+require './lib/solarwinds_apm/support/service_key_checker'
+
+# BUNDLE_GEMFILE=gemfiles/unit.gemfile bundle exec ruby -I test test/support/otlp_endpoint_test.rb
+describe 'OTLP Endpoint Test' do
+  before do
+    @original_env = ENV.to_h.dup
+    ENV.clear
+  end
+
+  after do
+    ENV.replace(@original_env)
+  end
+
+  # 2,5,8 (failed to auth token, not test here)
+  describe 'config_token' do
+    # 1
+    it 'wrong formatted SW_APM_SERVICE_KEY OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal false
+    end
+
+    # 3
+    it 'correct formatted SW_APM_SERVICE_KEY, successful auth token OTEL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'https://otel.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      _(endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      _(endpoint.instance_variable_get(:@service_name)).must_equal 'my_service'
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal true
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil)).must_equal 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+    end
+
+    # 4
+    it 'wrong formatted SW_APM_SERVICE_KEY APM PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal false
+    end
+
+    # 6
+    it 'correct formatted SW_APM_SERVICE_KEY, successful auth token APM PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      _(endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      _(endpoint.instance_variable_get(:@service_name)).must_equal 'my_service'
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal true
```

Comment:
this test case highlights why we really need assertions on the resolved OTLP endpoint